src/main/java/org/semux/consensus/SemuxBft.java

```diff
@@ -503,7 +502,7 @@ protected void onNewView(Proof p) {
         logger.trace("On new_view: {}", p);
 
         if (p.getHeight() == height // at same height
-                && p.getView() == view + 1 && state != State.COMMIT && state != State.FINALIZE) {// larger view
+                && p.getView() > view && state != State.COMMIT && state != State.FINALIZE) {// larger view
```

Comment:
The problem here was one validator proposing a view greater than current view with their validator was being taken over anothoher more valid block with lower view.  I think some protection here is needed.  But I have not personally tested through the logic, just saw testnet attacked in this way

Reply:
I believe the check logic is here https://github.com/semuxproject/semux-core/blob/40a00ccdace0f5ff5fe82b86e9a660f5ca758f39/src/main/java/org/semux/consensus/VoteSet.java#L72

Going to add some tests for it.